### PR TITLE
Stealth Transactions

### PR DESCRIPTION
**Pull request by cpacia**

### CODE REVIEW

**Comment by jackkleeman** [comment]
Is this aimed at vendors withdrawing money? Very cool.

**Comment by cpacia** [comment]
It allows someone to send bitcoins to another user's peerID. The coins arrive at a different bitcoin address each time preventing address reuse.

**Comment by jackkleeman** [comment]
Will this be used as part of the protocol or is it more of an extra useful feature?

**Comment by cpacia** [comment]
It's only going to be used in the wallet as far as I know.

**Comment by JustinDrake** [comment]
Can stealth transactions be used for direct OpenBazaar payments (i.e. no moderator)?

**Comment by cpacia** [comment]
Given that not all wallet implementations can handle stealth transactions (for example bitcoind can't) it's probably not a good idea.

File: repo/db/keys.go [hunk 1]
```diff
@@ -92,6 +111,25 @@ func (k *KeysDB) GetPathForScript(scriptPubKey []byte) (spvwallet.KeyPath, error
 	return p, nil
 }
 
+func (k *KeysDB) GetKeyForScript(scriptPubKey []byte) (*btcec.PrivateKey, error) {
+	k.lock.Lock()
+	defer k.lock.Unlock()
+
+	stmt, err := k.db.Prepare("select key from keys where scriptPubKey=? and purpose=-1")
```
Comment (tyler-smith) [inline repo/db/keys.go]: We should catch this to avoid a nil pointer panic on 121

**Review by ghost — CHANGES_REQUESTED** [review]
Just one small fix, but I think it should be made.

**Pull request merged**